implementation/src/main/java/com/teampc/controller/ViewSubmissionsController.java

```diff
@@ -57,32 +60,17 @@ public void setCurrentTest(Test test) {
       className.setText(currentTest.getCourseName());
 
       Collection<Submission> submissions = SubmissionDAO.getInstance().fetchSubmissionsForTest(currentTest);
-      Key key = currentTest.getKey();
-
-      // CANNED DATA
-      if (submissions.isEmpty()) {
-         Submission s = new Submission();
-         s.responses.add(new MultipleChoiceQuestionResponse());
-         s.responses.add(new MultipleChoiceQuestionResponse());
-         s.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
-         s.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
-         s.taker = UserSession.getLoggedInUser();
-         s.setTest(currentTest);
-
-         submissions = new ArrayList<>();
-         submissions.add(s);
-
-         key = new Key();
-         key.responses.add(new MultipleChoiceQuestionResponse());
-         key.responses.add(new MultipleChoiceQuestionResponse(1, new ArrayList<>()));
-         key.responses.add(new ShortAnswerQuestionResponse("", ShortAnswerQuestionResponse.MatchType.ALL));
-         key.responses.add(new ShortAnswerQuestionResponse("one", ShortAnswerQuestionResponse.MatchType.ALL));
-      }
+      Key key = currentTest.generateKey();
 
-      final Key keyPC = key;
-      submissions.stream().forEach(nextSubmission -> nextSubmission.gradeTest(keyPC));
+      if (key == null) {
```

Comment:
looks like this can't be null (based on the method implementation), so we can remove this check